packages/javascript-sdk/src/logger/index.ts

```diff
@@ -0,0 +1,47 @@
+import Config from '../config/index';
+
+interface LoggerFunctions {
+  warnFn?: (...msgs: unknown[]) => void;
+  errorFn?: (...msgs: unknown[]) => void;
+  logFn?: (...msgs: unknown[]) => void;
+}
+
+function customLogger() {
+  let frWarn = console.log,
+    frError = console.error,
+    frLog = console.log;
+
+  return {
+    addLogger({
+      logFn = console.log,
+      warnFn = console.warn,
+      errorFn = console.error,
+    }: LoggerFunctions) {
+      frLog = logFn;
+      frWarn = warnFn;
+      frError = errorFn;
+    },
+    shouldLog() {
+      return Config.logLevel !== 'none';
+    },
+    warn(...msgs: unknown[]) {
+      if (this.shouldLog()) {
+        frWarn(...msgs);
+      }
+    },
+    error(...msgs: unknown[]) {
+      if (this.shouldLog()) {
+        frError(...msgs);
+      }
+    },
+    log(...msgs: unknown[]) {
+      if (this.shouldLog()) {
+        frLog(...msgs);
+      }
+    },
+  };
+}
+
+const FRLogger = customLogger();
```

Comment:
Since this is no longer a class, should we start with a capital letter? Does it matter? Is there a difference?

Reply:
Probably not good convention so i can change this to lower case.